core/src/main/java/com/microsoft/applicationinsights/internal/statsbeat/StatsbeatHelper.java

```diff
@@ -0,0 +1,230 @@
+        INSTRUMENTATION_MAP_DECODING.put(43, "io.opentelemetry.javaagent.netty-4.0");
+        INSTRUMENTATION_MAP_DECODING.put(44, "io.opentelemetry.javaagent.netty-4.1");
+        INSTRUMENTATION_MAP_DECODING.put(45, "io.opentelemetry.javaagent.okhttp-3.0");
+        INSTRUMENTATION_MAP_DECODING.put(46, "io.opentelemetry.javaagent.opentelemetry-annotations-1.0");
+        INSTRUMENTATION_MAP_DECODING.put(47, "io.opentelemetry.javaagent.opentelemetry-api-1.0");
+        INSTRUMENTATION_MAP_DECODING.put(48, "io.opentelemetry.javaagent.reactor-3.1");
+        INSTRUMENTATION_MAP_DECODING.put(49, "io.opentelemetry.javaagent.reactor-netty-0.9");
+        INSTRUMENTATION_MAP_DECODING.put(50, "io.opentelemetry.javaagent.reactor-netty-1.0");
+        INSTRUMENTATION_MAP_DECODING.put(51, "io.opentelemetry.javaagent.servlet-2.2");
+        INSTRUMENTATION_MAP_DECODING.put(52, "io.opentelemetry.javaagent.servlet-3.0");
+        INSTRUMENTATION_MAP_DECODING.put(53, "io.opentelemetry.javaagent.servlet-common");
+        INSTRUMENTATION_MAP_DECODING.put(54, "io.opentelemetry.javaagent.spring-scheduling-3.1");
+        INSTRUMENTATION_MAP_DECODING.put(55, "io.opentelemetry.javaagent.spring-webflux-5.0");
+        INSTRUMENTATION_MAP_DECODING.put(56, "io.opentelemetry.javaagent.spring-webmvc-3.1");
+        INSTRUMENTATION_MAP_DECODING.put(57, "io.opentelemetry.javaagent.tomcat-7.0");
+
+        FEATURE_MAP = new HashMap<String, Integer>();
+        FEATURE_MAP.put(JAVA_VENDOR_ORACLE, 0);
+        FEATURE_MAP.put(JAVA_VENDOR_ZULU, 1);
+        FEATURE_MAP.put(JAVA_VENDOR_MICROSOFT, 2);
+        FEATURE_MAP.put(JAVA_VENDOR_ADOPT_OPENJDK, 3);
+        FEATURE_MAP.put(JAVA_VENDOR_REDHAT, 4);
+        FEATURE_MAP.put(JAVA_VENDOR_OTHER, 5);
+
+        FEATURE_MAP_DECODING = new HashMap<Integer, String>();
+        FEATURE_MAP_DECODING.put(0, JAVA_VENDOR_ORACLE);
+        FEATURE_MAP_DECODING.put(1, JAVA_VENDOR_ZULU);
+        FEATURE_MAP_DECODING.put(2, JAVA_VENDOR_MICROSOFT);
+        FEATURE_MAP_DECODING.put(3, JAVA_VENDOR_ADOPT_OPENJDK);
+        FEATURE_MAP_DECODING.put(4, JAVA_VENDOR_REDHAT);
+        FEATURE_MAP_DECODING.put(5, JAVA_VENDOR_OTHER);
+    }
+
+    public static long encodeInstrumentations(Set<String> instrumentations) {
+        return encode(instrumentations, INSTRUMENTATION_MAP);
+    }
+
+    public static long encodeFeature(Set<String> features) {
+        return encode(features, FEATURE_MAP);
+    }
+
+    public static Set<String> decodeInstrumentations(long num) {
+        return decode(num, INSTRUMENTATION_MAP_DECODING);
+    }
+
+    public static Set<String> decodeFeature(long num) {
+        return decode(num, FEATURE_MAP_DECODING);
+    }
+
+    private static long encode(Set<String> list, Map<String, Integer> map) {
+        long number = 0L;
+        for (String item : list) {
+            int index = map.get(item);
+            number |= getPowerOf2(index);
+        }
+
+        return number;
```

Comment:
kusto cluster doesn't support BitSet.. what is the benefit using BitSet?

Reply:
it's going to do more efficient (bitwise) powers of 2, and then you can get the long back from it at the end, it's designed for this kind of thing, esp since we may want 2 longs in the future